File: ClientApp/PacClientApp/areaInformationManagementIF/area.cpp
#include "area.h"

namespace areaInformationManagementIF {
Area::Area(QObject *parent)
    : helpers::EntityIF{}
{
    setParent(parent);
}

QJsonDocument Area::toJsonDocument()
{
    QJsonObject areaObject;
    areaObject.insert("idArea", idArea);
    areaObject.insert("areaName", areaName);
    areaObject.insert("lat", lat);
    areaObject.insert("lng", lng);
    areaObject.insert("istatCode", istatCode);

    return QJsonDocument(areaObject);
}

void Area::fromJsonObject(QJsonObject jsonObject)
{
    idArea = jsonObject.value("idArea").toInt();
    areaName = jsonObject.value("areaName").toString();
    lat = jsonObject.value("lat").toDouble();
    lng = jsonObject.value("lng").toDouble();
    istatCode = jsonObject.value("istatCode").toString();
}

int Area::getIdArea()
{
    return idArea;
}

void Area::setIdArea(int newIdArea)
{
    idArea = newIdArea;
}

QString Area::getAreaName()
{
    return areaName;
}

void Area::setAreaName(const QString &newAreaName)
{
    areaName = newAreaName;
}

double Area::getLat()
{
    return lat;
}

void Area::setLat(double newLat)
{
    lat = newLat;
}

double Area::getLng()
{
    return lng;
}

void Area::setLng(double newLng)
{
    lng = newLng;
}

QString Area::getIstatCode()
{
    return istatCode;
}

void Area::setIstatCode(const QString &newIstatCode)
{
    istatCode = newIstatCode;
}

}

File: WebServer/LVSEmergency/src/main/java/it/lvsemergency/areaInformationManagement/Area.java
package it.lvsemergency.areaInformationManagement;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.Table;

@Table(name = "area")
@Entity
public class Area {
	private @Id @GeneratedValue(strategy = GenerationType.IDENTITY) Integer idArea;

	private String areaName;
	private Double lat;
	private Double lng;
	private String nameAprsStation;
	private String istatCode;

	public Area() {

	}

	public Area(Integer idArea, Double lat, Double lng, String nameAprsStation, String istatCode) {
		super();
		this.idArea = idArea;
		this.lat = lat;
		this.lng = lng;
		this.nameAprsStation = nameAprsStation;
		this.istatCode = istatCode;
	}

	public Integer getIdArea() {
		return idArea;
	}

	public void setIdArea(Integer idArea) {
		this.idArea = idArea;
	}

	public String getAreaName() {
		return areaName;
	}

	public void setAreaName(String areaName) {
		this.areaName = areaName;
	}

	public Double getLat() {
		return lat;
	}

	public void setLat(Double lat) {
		this.lat = lat;
	}

	public Double getLng() {
		return lng;
	}

	public void setLng(Double lng) {
		this.lng = lng;
	}
	
	public String getNameAprsStation() {
		return nameAprsStation;
	}
	
	public void setNameAprsStation(String nameAprsStation) {
		this.nameAprsStation = nameAprsStation;
	}

	public String getIstatCode() {
		return istatCode;
	}

	public void setIstatCode(String istatCode) {
		this.istatCode = istatCode;
	}
}
